src/components/organisms/topbar/project/manualRun/manualRunParamsForm.tsx

```diff
@@ -1,112 +1,243 @@
-		append({ key: "", value: "" });
+	const { projectId } = useParams();
+
+	const { projectManualRun, updateManualRunConfiguration } = useManualRunStore((state) => ({
+		projectManualRun: state.projectManualRun[projectId!],
+		updateManualRunConfiguration: state.updateManualRunConfiguration,
+	}));
+
+	const { isJson } = projectManualRun || {};
+
+	const convertParamsToJson = (currentParams: { key: string; value: string }[]) => {
+		const jsonObject = Object.fromEntries(
+			currentParams.map((param: { key: string; value: any }) => {
+				try {
+					return [param.key, JSON.parse(param.value)];
+				} catch {
+					return [param.key, param.value];
+				}
+			})
+		);
+		setValue("jsonParams", JSON.stringify(jsonObject, null, 2), { shouldValidate: true });
 	};
 
-	const errors = formState.errors?.params as FieldErrors<FieldValues> & ManualFormParamsErrors;
+	useEffect(() => {
+		if (isJson) {
+			const currentParams = getValues("params");
+			convertParamsToJson(currentParams);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [projectId]);
+
+	const [useJsonEditor, setUseJsonEditor] = useState(isJson);
+
+	const errors = formState.errors as FieldErrors<FieldValues> & ManualFormParamsErrors;
+
+	const validateParams = () => {
+		const params = getValues("params");
+		let isValid = true;
+
+		params.forEach((param: { key: string; value: string }, index: number) => {
+			if (!param.key.trim()) {
+				setError(`params.${index}.key`, {
+					type: "manual",
+					message: t("keyIsRequired"),
+				});
+				isValid = false;
+			}
+			if (!param.value.trim()) {
+				setError(`params.${index}.value`, {
+					type: "manual",
+					message: t("valueIsRequired"),
+				});
+				isValid = false;
+			}
+		});
+
+		return isValid;
+	};
 
-	return (
-		<form className="mt-9">
-			<div className="mb-4 flex items-center gap-1 text-base text-gray-500">
-				{t("titleParams")}
+	const handleAddParam = () => {
+		if (!fields.length || validateParams()) {
+			append({ key: "", value: "" });
+		}
+	};
 
-				<div className="cursor-pointer" title={t("titleParams")}>
-					<InfoIcon className="fill-white" />
+	const handleFieldChange = (index: number, field: "key" | "value", value: string) => {
+		setValue(`params.${index}.${field}`, value, { shouldValidate: true });
+		trigger(`params.${index}.${field}`);
+	};
+
+	const handleJsonChange = (value?: string) => {
+		setValue("jsonParams", value, { shouldValidate: true });
+		if (value) {
+			try {
+				const parsedJson = JSON.parse(value);
+				const newParams = Object.entries(parsedJson).map(([key, value]) => {
+					if (typeof value === "object") return { key, value: JSON.stringify(value) };
+
+					return { key, value: String(value) };
+				});
+				setValue("params", newParams, { shouldValidate: false });
+			} catch {
+				// If JSON is invalid, don't update params
+			}
+		} else {
+			setValue("params", [], { shouldValidate: false });
+		}
+	};
+
+	const toggleEditorMode = () => {
```

Comment:
I don't like that it changes from JSON to inputs , I see an empty `""`
![image](https://github.com/user-attachments/assets/623dd085-87bf-4d1f-a44a-0b72a8ef17b5)
![image](https://github.com/user-attachments/assets/ddce4c3a-69e1-4f24-80e9-c36ce3e6131f)

Reply:
fixed